### PR TITLE
Fix !ts hat generation for overlapping ranges

### PR DESCRIPTION
**Pull request by tjjfvi**

### CODE REVIEW

**Comment by robertt** [comment]
Do you have an example of where this would be an issue?

**Comment by tjjfvi** [comment]
Yep: <https://discord.com/channels/508357248330760243/755409256290713681/945416717927055360>

**Comment by robertt** [comment]
cool! just making sure, tested?

File: src/modules/twoslash.ts [hunk 1]
```diff
@@ -152,18 +152,44 @@ export class TwoslashModule extends Module {
...
+					// map to [start, end (non-inclusive)]
+					.map(
+						error =>
+							[
+								error.character!,
+								error.character! + error.length!,
+							] as const,
+					)
+					// sort by start, ascending
+					.sort((a, b) => a[0] - b[0])
```
Comment (ckiee) [inline src/modules/twoslash.ts:168]: Could this sort be moved before the `map` for a bit more clarity?

File: src/modules/twoslash.ts [hunk 2]
```diff
@@ -152,18 +152,44 @@ export class TwoslashModule extends Module {
...
+					// fix overlapping ranges
```
Comment (ckiee) [inline src/modules/twoslash.ts:169]: "fix"? Can the comment elaborate?

**Review by ckiee — CHANGES_REQUESTED** [review]
Nitpicks

File: src/modules/twoslash.ts (hunk 1 above)
Comment (tjjfvi) [reply src/modules/twoslash.ts:168]: It could, but more `!`s would be needed

File: src/modules/twoslash.ts (hunk 2 above)
Comment (tjjfvi) [reply src/modules/twoslash.ts:169]: Not sure how to get more specific than "fix" without repeating the implementation

**Review by robertt — APPROVED** [review]

**Pull request merged**